### PR TITLE
removed INJECTION const that when injected affected certain styles on…

### PR DESCRIPTION
**Pull request by samholst**
… page. Extracted reload script to its own js folder to resolve css issues. Added in condition to load script if in development in application layout and routes.

### Description of the Change

Amber reload was previously removed due to unknown issues and at least for the projects I worked on in Crystal, it would inadvertently skew certain css stylings when added on the the `context.response`. By extracting out the JavaScript file into in the public folder, it fixed the issues I was having with the previous code. Due to the extraction, a static socket route is need in order for the JavaScript to know what socket to connect to.

### Alternate Designs

Could possibly get a shard that is included by default in amber instead of a pipeline.

### Benefits

Codebase and css styling functioned as wanted and were not effected by extra padding and positioning from the injected constant that was previously used.

### Possible Drawbacks

May not qualify as a pipeline with the new changes.

### CODE REVIEW

**Comment by drujensen** [comment]
@samholst This is great!  thanks for working on this.  💯 Is it ready to be reviewed?

**Comment by samholst** [comment]
@drujensen Thanks! In just a few it should be ready for review :)

**Comment by elorest** [comment]
@faustinoaq was working on something similar here: https://github.com/amberframework/amber/pull/865 

@drujensen @faustinoaq Perhaps we should include the JS file optionally based upon a variable set or added by the `reload_pipe` to context somehow.

File: src/amber/pipes/reload.cr [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require "../../support/client_reload"
```
[the remark this answers is not in the document]
Comment (elorest) [reply src/amber/pipes/reload.cr]: Looks good.

**Comment by samholst** [comment]
@elorest @drujensen Ready for review :)

File: src/amber/cli/templates/app/src/{{name}}.cr [hunk 2]
```diff
@@ -1,3 +1,4 @@
 require "../config/*"
 
+Amber::Support::ClientReload.new if Amber.settings.auto_reload?
```
[the remark this answers is not in the document]
Comment (samholst) [reply src/amber/cli/templates/app/src/{{name}}.cr]: I'm for that. Let me make that change real fast.

File: src/amber/cli/templates/app/config/initializers/client_reload.cr [hunk 3]
```diff
@@ -0,0 +1 @@
+Amber::Support::ClientReload.new if Amber.settings.auto_reload?
```
Comment (drujensen) [inline src/amber/cli/templates/app/config/initializers/client_reload.cr]: 💯

**Comment by drujensen** [comment]
@elorest can we merge this?

File: src/amber/cli/templates/app/config/environments/development.yml.ecr [hunk 4]
```diff
@@ -29,6 +29,7 @@ database_url: postgres://postgres:@localhost:5432/<%= database_name_base %>_deve
 <% when "sqlite" -%>
 database_url: sqlite3:./db/<%= database_name_base %>_development.db
 <% end -%>
+auto_reload: true
```
[the remark this answers is not in the document]
Comment (elorest) [reply src/amber/cli/templates/app/config/environments/development.yml.ecr]: <3

File: src/amber/cli/templates/app/config/initializers/client_reload.cr (hunk 3 above)
Comment (drujensen) [reply src/amber/cli/templates/app/config/initializers/client_reload.cr]: I disagree.  This is not really core to the framework and is an add-on.  It should be easily removed.  To your point, maybe the setting should be moved to the custom settings?

**Comment by drujensen** [comment]
I talked to @elorest and we will go with an initializer for now in anticipation for making live_reload an add-on.

**Pull request merged**